src/sentry/workflow_engine/handlers/action/notification/handler.py

```diff
@@ -83,15 +86,77 @@ def execute(
         action: Action,
         detector: Detector,
     ) -> None:
-        try:
-            handler = group_type_notification_registry.get(detector.type)
-            handler.handle_workflow_action(job, action, detector)
-        except NoRegistrationExistsError:
-            logger.exception(
-                "No notification handler found for detector type: %s",
-                detector.type,
-                extra={"detector_id": detector.id, "action_id": action.id},
-            )
+        execute_via_group_type_registry(job, action, detector)
```

Comment:
If you want to make this even moar SOILD, you could define the `execute` interface on the ABC then invoke the `execute_via_group_type_registry` in each class. it's a bit of replication, but it would be how you uphold the `I` in solid (inversion principle).

that said, i don't think we'd gain much value from doing that work unless we plan to start overriding this method in the individual action handlers (we could do this refactor when that use case arises), just wanted to call out the example. :)